do.rb

```diff
@@ -1,29 +1,29 @@
 #!/usr/bin/env ruby
+require 'rake'
 require 'json'
-require 'fileutils'
+require 'time'
 
 Dir.chdir(File.dirname(__FILE__))
 
 json = { data: {} }
 data = json[:data]
-Dir.entries('test').each do |path|
-  next if path == '.' or path == '..'
-  user, repo, _ = path.split('.')
+Dir.glob('test/*.sh').each do |path|
+  user, repo, _ = File.basename(path).split('.')
   puts cmd = "git clone https://github.com/#{user}/#{repo} --depth=1 -b master"
   `#{cmd}`
   Dir.chdir(repo) do
-    `sh ../test/#{path}`
+    sh "sh ../#{path}"
```

Comment:
If you want to use `sh "../#{path}"` instead of `sh "sh ../#{path}"`, you should insert `#!/bin/sh` to the first line of the script.